odoo/addons/base/models/res_users.py

```diff
@@ -225,7 +225,7 @@ def write(self, vals):
         # DLE P139
         if self.ids:
             self.env['ir.model.access'].call_cache_clearing_methods()
-            self.env['res.users'].has_group.clear_cache(self.env['res.users'])
+            self.env['res.users']._has_group.clear_cache(self.env['res.users'])
```

Comment:
ORM caches were unified into a single LRU a while back, so this should just be
```suggestion
            self.env.registry._clear_cache()
```

Although in reality `call_cache_clearing_methods` is *already* clearing the ormcache LRU. I guess it doesn't hurt to make sure...

Reply:
I've add a unit test. I haven't found a clean solution to test that cache is invalidated before calling write on it's parent as I supposed cache is cleared afterwards anyway. But according the unit-test call_cache_clearing_methods properly invalidate _has_group cache.

Reply:
@xmo-odoo, I'm wondering if that comment above in `res.groups` write is still present in v14.0 ?

```python
        # invalidate caches before updating groups, since the recomputation of
        # field 'share' depends on method has_group()
        # DLE P139
```

Reply:
> @xmo-odoo, I'm wondering if that comment above in res.groups write is still present in v14.0 ?

It might not as @Feyensv apparently updated `_compute_share` to stop using `has_group` in 0279cbc4a79162ea837bf14db73e12b38cd0aca9

So it's possible that we could remove not just the comment but the cache clearing entirely, @rco-odoo what do you think?

> And is it coverd by a test some how ?

Unlikely.